modules/windows-function/locals.tf

```
@@ -1,8 +1,9 @@
 locals {
-  is_consumption = data.azurerm_service_plan.plan.sku_name == "Y1"
+  is_consumption     = contains(["Y1"], data.azurerm_service_plan.plan.sku_name)
+  is_elastic_premium = contains(["EP1", "EP2", "EP3"], data.azurerm_service_plan.plan.sku_name)
```

Elastic and consumption are the same thing and is only used for `always_on`.

```suggestion
  is_consumption     = contains(["Y1", "EP1", "EP2", "EP3"], data.azurerm_service_plan.plan.sku_name)
```

---

those SKUs are not consumption plan, which is why I put them into new local `is_elastic_premium`;
Moreover, in future, there might be other limitations with regards to EP plans compared to Y plans, so I think it is better to keep them separated;
What do you think @BzSpi  ?

---

They are clearly not the same: https://learn.microsoft.com/en-us/azure/azure-functions/functions-premium-plan?tabs=portal